tensorboard/webapp/widgets/linked_time_fob/types.ts

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {PluginType} from '../../metrics/types';
+import {TemporalScale} from '../histogram/histogram_types';
+import {Scale} from '../line_chart_v2/lib/public_types';
+
+export {Scale} from '../line_chart_v2/lib/public_types';
+
+export enum AxisDirection {
```

Comment:
There seems to be a 1:1 relationship with Axis and PluginType at the moment - so maybe FobData should just contain AxisDirection?

Reply:
I am trying to keep things generalized so that this can be used for other cards(like the image card) if we decide to. Note that the data from each card is different so we still need a type that can encompass both sets of data and ideally one that enforces that we have all of one or all of the other.

Reply:
I think what I'm saying is that you have not generalized it sufficiently.  You have this:

```
  public getAxisDirection() {
    return this.fobCardData.type === PluginType.SCALARS
      ? AxisDirection.HORIZONTAL
      : AxisDirection.VERTICAL;
  }
```

To be truly "generalized" I would expect AxisDirection to be a field in FobCardData and the value of AxisDirection would not depend on PluginType so explicitly.